inc/namespace.php

```diff
@@ -88,69 +93,98 @@ function ajax_select() : void {
 			continue;
 		}
 
-		$mime_type = $selection['mime'];
-
-		$args = [
-			'post_title' => $selection['title'],
-			'post_parent' => $post_id,
-			'post_name' => $selection['id'],
-			'post_content' => $selection['description'],
-			'post_excerpt' => $selection['caption'],
-			'post_mime_type' => $mime_type,
-			'guid' => $selection['url'],
-			'meta_input' => $selection['meta'],
-		];
+		try {
+			$media = get_media_from_selection( $selection );
+			$attachment = insert_attachment( $media, $provider, $post_id );
+		} catch ( Exception $e ) {
+			wp_send_json_error( $e->getMessage() );
+		}
 
-		$attachment_id = wp_insert_attachment( $args, false, 0, true );
+		$attachments[ $selection['id'] ] = $attachment->ID;
+	}
 
-		if ( is_wp_error( $attachment_id ) ) {
-			wp_send_json_error( $attachment_id );
-		}
+	wp_send_json_success( $attachments );
+}
 
-		if ( ! empty( $selection['alt'] ) ) {
-			add_post_meta( $attachment_id, '_wp_attachment_image_alt', wp_slash( $selection['alt'] ) );
-		}
+function get_media_from_selection( array $selection ) : Media {
+	$media = new Media( $selection['id'], $selection['mime'] );
 
-		$metadata = wp_get_attachment_metadata( $attachment_id, true );
-		if ( ! is_array( $metadata ) ) {
-			$metadata = [];
-		}
+	foreach ( $selection as $key => $value ) {
+		$media->{ $key } = $value;
+	}
 
-		$metadata['file'] = wp_slash( $selection['filename'] );
+	return $media;
+}
 
-		if ( ! empty( $selection['width'] ) ) {
-			$metadata['width'] = intval( $selection['width'] );
-		}
+function insert_attachment( Media $media, Provider $provider, int $parent = null ) : WP_Post {
+	global $amf_inserted;
```

Comment:
Is there any way around using this global?

Reply:
Noted below - this isn't needed any more so yes.